src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -61,4 +67,194 @@ default GameProfile identity() {
+     *
+     * @return The current level the player has
+     */
+    default Value.Mutable<Integer> experienceLevel() {
+        return this.requireValue(Keys.EXPERIENCE_LEVEL).asMutable();
+    }
+
+    /**
+     * {@link Keys#EXPERIENCE_SINCE_LEVEL}
+     *
+     * @return The amount of experience the player has collected towards the next level
+     */
+    default Value.Mutable<Integer> experienceSinceLevel() {
+        return this.requireValue(Keys.EXPERIENCE_SINCE_LEVEL).asMutable();
+    }
+
+    /**
+     * {@link Keys#FLYING_SPEED}
+     *
+     * @return The speed at which the player flies
+     */
+    default Value.Mutable<Double> flyingSpeed() {
+        return this.requireValue(Keys.FLYING_SPEED).asMutable();
+    }
+
+    /**
+     * {@link Keys#FOOD_LEVEL}
+     *
+     * @return The player's food level
+     */
+    default Value.Mutable<Integer> foodLevel() {
+        return this.requireValue(Keys.FOOD_LEVEL).asMutable();
+    }
+
+    /**
+     * {@link Keys#IS_FLYING}
+     *
+     * @return Whether the entity is flying
+     */
+    default Value.Mutable<Boolean> flying() {
+        return this.requireValue(Keys.IS_FLYING).asMutable();
+    }
+
+    /**
+     * {@link Keys#IS_SLEEPING}
+     *
+     * @return Whether the player is sleeping
+     */
+    default Value<Boolean> sleeping() {
+        return this.requireValue(Keys.IS_SLEEPING);
+    }
+
+    /**
+     * {@link Keys#IS_SLEEPING_IGNORED}
+     *
+     * @return Whether this player is going to be ignored for sleeping to "reset" the day
+     */
+    default Value<Boolean> sleepingIgnored() {
+        return this.requireValue(Keys.IS_SLEEPING_IGNORED);
+    }
+
+    /**
+     * {@link Keys#MAX_EXHAUSTION}
+     *
+     * @return The maximum exhaustion of the player
+     */
+    default Value<Double> maxExhaustion() {
+        return this.requireValue(Keys.MAX_EXHAUSTION);
+    }
+
+    /**
+     * {@link Keys#MAX_FOOD_LEVEL}
+     *
+     * @return The maximum food level of the player
+     */
+    default Value<Integer> maxFoodLevel() {
+        return this.requireValue(Keys.MAX_FOOD_LEVEL);
+    }
+
+    /**
+     * {@link Keys#MAX_SATURATION}
+     *
+     * @return The maximum saturation of the player
+     */
+    default Value<Double> maxSaturation() {
+        return this.requireValue(Keys.MAX_SATURATION);
+    }
+
+    /**
+     * {@link Keys#SATURATION}
+     *
+     * @return The player's saturation
+     */
+    default Value.Mutable<Double> saturation() {
+        return this.requireValue(Keys.SATURATION).asMutable();
+    }
+
+    /**
+     * {@link Keys#SLEEP_TIMER}
+     *
+     * @return The sleep timer of the player
+     */
+    default Value.Mutable<Integer> sleepTimer() {
+        return this.requireValue(Keys.SLEEP_TIMER).asMutable();
+    }
+
+    /**
+     * {@link Keys#FIRST_DATE_JOINED}
+     *
+     * @return The timestamp value when this player first joined
+     */
+    default Optional<Value.Mutable<Instant>> firstJoined() {
```

Comment:
This should be on `ServerPlayer`

`Player` can also be a `ClientPlayer` which will not know this information.

Reply:
https://github.com/SpongePowered/Sponge/blob/1d33b3c3daf2fbcee44554d128deec16c11aa96e/src/main/java/org/spongepowered/common/data/provider/entity/IdentifiableData.java#L47-L54

Reply:
That supports line is wrong, thanks for bringing this to our attention. Should be `h instanceof ServerPlayer`